src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,178 @@
 import json
 import logging
+from typing import Dict, List
 
 import d3a.constants
 from d3a.d3a_core.exceptions import InvalidBidOfferPair
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
-from d3a.models.market import validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import (
-    BidOfferMatch, offer_or_bid_from_json_string)
+from d3a.models.market import Market
+from d3a.models.market.market_structures import BidOfferMatch
 
 from d3a.models.myco_matcher.base_matcher import BaseMatcher
 
 
 class ExternalMatcher(BaseMatcher):
+    """Class responsible for external bids / offers matching."""
     def __init__(self):
         super().__init__()
         self.simulation_id = d3a.constants.COLLABORATION_ID
         self.myco_ext_conn = None
         self.channel_prefix = f"external-myco/{self.simulation_id}/"
+        self.response_channel = f"{self.channel_prefix}/response"
+        self.events_channel = f"{self.response_channel}/events/"
         self._setup_redis_connection()
         self.area_uuid_markets_mapping = {}
-        self.markets_mapping = {}
+        self.markets_mapping = {}  # Dict[market_id: market] mapping
         self.recommendations = []
 
     def _setup_redis_connection(self):
         self.myco_ext_conn = ResettableCommunicator()
         self.myco_ext_conn.sub_to_multiple_channels(
-            {"external-myco/get-simulation-id": self.get_simulation_id,
+            {"external-myco/get-simulation-id": self.publish_simulation_id,
```

Comment:
IMO `get_simulation_id` is better even for the callback method (i was confused at first)

Reply:
I'm trying to homogenize the callbacks, the ones that publish to redis are appended with `publish`

Reply:
I got that, but still. Lets wait for other opinions.

Reply:
Actually, I'm thinking that using `get-*` or `publish-*` in channel names is not the best thing. The reason is that you always have two sides of the communication that do opposite things (pub/sub) so if you use `get/publish/subscribe` in the channel name you're only covering one part of the story.
I would personally use something like `external-myco/simulation-id`

Reply:
The same applies to `post-recommendations`: only one side is posting them, while the other side is receiving them.